### PR TITLE
feat(backend): Verify Derivation Origin's Id Alias from POUH Credential

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Fix the issue in the POUH credential which didn't check the derivation origin of the POUH credential.

# Changes

* Upgrade `ic-verifiable-credentials` which has the logic to verify the new parameter.
* Expect the new parameter in backend endpoint.
* Update candid and generate related files.
* Pass the derivation origin or the window origin when adding the credential.

# Tests

Tested locally and it works.
Fix tests.

### CODE REVIEW

**Comment by lmuntaner** [comment]
@peterpeterparker @AntonioVentilii please review

File: src/backend/tests/it/utils/mock.rs [hunk 1]
```diff
@@ -13,10 +13,11 @@ pub const WEENUS_SYMBOL: &str = "Weenus";
...
+pub const VC_HOLDER: &str = "7eboi-tyuys-aqm4c-w2l7i-vgucm-xvawx-lemzx-6kq2g-f53u7-yvfh2-nae";
 pub const ISSUER_CANISTER_ID: &str = "qdiif-2iaaa-aaaap-ahjaq-cai";
 pub const ISSUER_ORIGIN: &str = "https://dummy-issuer.vc/";
-pub const II_CANISTER_ID: &str = "rdmx6-jaaaa-aaaaa-aaadq-cai";
+pub const VC_DERIVATION_ORIGIN: &str = "https://l7rua-raaaa-aaaap-ahh6a-cai.ic0.app";
+pub const II_CANISTER_ID: &str = "fgte5-ciaaa-aaaad-aaatq-cai";
```
Comment (AntonioVentilii) [inline src/backend/tests/it/utils/mock.rs:21]: out of curiosity, why the II canister changed too for the mocks?

Comment (lmuntaner) [reply src/backend/tests/it/utils/mock.rs:21]: Because I created the JWT with a different II.

Comment (AntonioVentilii) [reply src/backend/tests/it/utils/mock.rs:21]: Ah understood! tks for the explanation!

Comment (lmuntaner) [reply src/backend/tests/it/utils/mock.rs:21]: Are you going to approve it? Or should I ask for @peterpeterparker 's review specifically?

Comment (AntonioVentilii) [reply src/backend/tests/it/utils/mock.rs:21]: honestly, to me it looks good, it seems trivial, but for this I would prefer to rely on @peterpeterparker or @bitdivine to review

**Review by peterpeterparker — COMMENTED** [review]
Can you provide a frontend test for `requestPouchCredential` service?

I believe it hasn't been covered yet. Ideally, providing the test before this PR, so that the PR explicitly extends it with the changes. However, if we're under time pressure, providing it afterward for sure is also fine.

**Comment by peterpeterparker** [comment]
I went running and give it some more though, so I would have two more questions (😅):

- I know what's the `derivation_origin` of II, but, what's the `derivation_origin` here? Is there any specification about it?

- Is it actually secure to pass this information from the frontend to backend?

File: src/backend/backend.did [hunk 2]
```diff
@@ -5,6 +5,7 @@ type AddUserCredentialError = variant {
   UserNotFound;
 };
 type AddUserCredentialRequest = record {
+  derivation_origin : text;
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply src/backend/backend.did:8]: Yes, it's necessary for the new VC SDK and that is precisely what security asked for.

File: src/frontend/src/lib/services/request-pouh-credential.services.ts [hunk 3]
```diff
@@ -112,6 +118,8 @@ export const requestPouhCredential = ({
 }): Promise<ResultSuccess> => {
 	const credentialSubject = identity.getPrincipal();
 	const { auth: authI18n } = get(i18n);
+	const derivationOriginParams = getOptionalDerivationOrigin();
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply src/frontend/src/lib/services/request-pouh-credential.services.ts:121]: Sure, I can do it later.

**Comment by lmuntaner** [comment]
> * I know what's the `derivation_origin` of II, but, what's the `derivation_origin` here? Is there any specification about it?

That is percisely the same concept as the `derivation_origin` in II. That's why I used the same function for logging in. The difference is that when logging in, if not provded, it uses the window origin. In here, we need to be specific of which one.

> * Is it actually secure to pass this information from the frontend to backend?

I don't see any problem. But I will ask security to review this PR.

**Comment by lmuntaner** [comment]
Closing this PR because it's a security issue to pass the derivation origin from the frontend.

**Pull request closed without merging**